Tests/Extractor/FrontendExtractorTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Bazinga\JsTranslationBundle\Tests\Extractor;
+
+use Bazinga\Bundle\JsTranslationBundle\Extractor\FrontendExtractor;
+use Bazinga\Bundle\JsTranslationBundle\Tests\WebTestCase;
+use Symfony\Component\Translation\MessageCatalogue;
+
+final class FrontendExtractorTest extends WebTestCase
+{
+    const TEST_LOCALE = 'en';
+    const TEST_KEY_1 = 'test-key-1';
+
+    /**
+     * @var FrontendExtractor
+     */
+    protected $extractor;
+
+    public function setUp()
+    {
+      $container = $this->getContainer();
+      $this->extractor = $container->get('bazinga.jstranslation.translation_frontend_extractor');
+    }
+
+    /**
+     * @dataProvider resourcesWithNotValidTransFunctionUsage
+     */
+    public function testExtractShouldNotRetrieveTransKey($resources)
+    {
+        $catalogue = new MessageCatalogue(self::TEST_LOCALE);
+        $this->extractor->extract($resources, $catalogue);
+        $this->assertEmpty($catalogue->all());
+    }
+
+    /**
+     * @dataProvider resourcesWithATransFunctionUsage
+     */
+    public function testExtractShouldRetrieveTransKey($resources)
+    {
+        $catalogue = new MessageCatalogue(self::TEST_LOCALE);
+        $this->extractor->extract($resources, $catalogue);
+        $this->assertTrue($catalogue->has(self::TEST_KEY_1));
+    }
+
+    /**
+     * @dataProvider resourcesWithNotValidTransChoiceFunctionUsage
+     */
+    public function testExtractShouldNotRetrieveTransChoiceKey($resources)
+    {
+        $catalogue = new MessageCatalogue(self::TEST_LOCALE);
+        $this->extractor->extract($resources, $catalogue);
+        $this->assertEmpty($catalogue->all());
+    }
+
+    /**
+     * @dataProvider resourcesWithATransChoiceFunctionUsage
```

Comment:
no need to create data providers in case you have a single resource to process. Put the name of the file directly in the test.

Reply:
OK